src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs

```diff
@@ -205,9 +206,14 @@ internal void EndGetConnection(IAsyncResult result)
                 throw new ArgumentNullException(nameof(recipients));
             }
 
-            if (GlobalLog.IsEnabled && recipients.Count > 0)
+            if (recipients.Count <= 0)
```

Comment:
Yeah this is populated by _recipients, and we do throw in SmtpClient, if no recipient information was given. Will remove these asserts.